packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -3,29 +3,87 @@ import { assert, details } from '@agoric/assert';
 import makeStore from '@agoric/store';
 import makeWeakStore from '@agoric/weak-store';
 import makeAmountMath from '@agoric/ertp/src/amountMath';
+import { makeTable, makeValidateProperties } from '@agoric/zoe/src/table';
+import { E } from '@agoric/eventual-send';
 
 import makeObservablePurse from './observable';
 import makeOfferCompiler from './offer-compiler';
+import { makeDehydrator } from './lib-dehydrate';
 
 // does nothing
 const noActionStateChangeHandler = _newState => {};
 
-export async function makeWallet(
-  E,
+export async function makeWallet({
   zoe,
+  // eslint-disable-next-line no-unused-vars
+  mailboxAdmin,
+  // eslint-disable-next-line no-unused-vars
+  board,
   registry,
   pursesStateChangeHandler = noActionStateChangeHandler,
   inboxStateChangeHandler = noActionStateChangeHandler,
-) {
-  const petnameToPurse = makeStore();
-  const purseToIssuer = makeWeakStore();
-  const issuerPetnameToIssuer = makeStore();
+}) {
+  // Create the petname maps so we can dehydrate information sent to
+  // the frontend.
+  const { makeMapping } = makeDehydrator();
+  const purseMapping = makeMapping('purse');
+  const brandMapping = makeMapping('brand');
+
+  // Brand Table
+  // Columns: key:brand | brand | issuer | amountMath
```

Comment:
I think it's a fine improvement to label the key, but this doesn't need to repeat `brand` as the first column, too.

Reply:
Good point, I'll change that